pkg/analysis/analysis_test.go

```diff
@@ -14,14 +14,127 @@ limitations under the License.
 package analysis
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
-	"testing"
-
 	"github.com/k8sgpt-ai/k8sgpt/pkg/common"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/kubernetes"
+	"github.com/magiconair/properties/assert"
+	"github.com/spf13/viper"
 	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"strings"
+	"testing"
 )
 
+// sub-function
+func analysis_RunAnalysis_Filter_Tester(t *testing.T, filterFlag string) []common.Result {
```

Comment:
Can we please make the function signatures snake_case lower cased?

Reply:
fixed, thanks